lib/scraped/response/decorator/absolute_urls.rb

```diff
@@ -7,23 +7,31 @@ class Decorator
       class AbsoluteUrls < Decorator
         def body
           Nokogiri::HTML(super).tap do |doc|
+            @doc = doc
             doc.css('img').each { |img| img[:src] = absolute_url(img[:src]) }
             doc.css('a').each { |a| a[:href] = absolute_url(a[:href]) }
           end.to_s
         end
 
         private
 
+        attr_reader :doc
+
         def absolute_url(relative_url)
           unless relative_url.to_s.empty?
-            URI.join(url, URI.encode(
+            URI.join(base_url, URI.encode(
               # To prevent encoded URLs from being encoded twice
               URI.decode(relative_url)
             ).gsub('[', '%5B').gsub(']', '%5D')).to_s
           end
         rescue URI::InvalidURIError
           relative_url
         end
+
+        def base_url
+          return url if (base = doc.css('base @href').text).empty?
```

Comment:
There's a little bit too much going on in this line. I think I'd simplify this to something like:

```ruby
base = doc.at_css('base @href') or return url
base.text
```